porch/repository/pkg/git/testing_repo.go

```diff
@@ -61,12 +68,58 @@ func InitEmptyRepositoryWithWorktree(t *testing.T, path string) *gogit.Repositor
 	return repo
 }
 
+func ServeGitRepositoryWithBranch(t *testing.T, tarfile, tempdir, branch string) (*gogit.Repository, string) {
+	t.Helper()
+
+	git := OpenGitRepositoryFromArchive(t, tarfile, tempdir)
+	InitializeBranch(t, git, branch)
+	return git, ServeExistingRepository(t, git)
+}
+
+func InitializeBranch(t *testing.T, git *gogit.Repository, branch string) {
+	t.Helper()
+
+	// If main branch exists, rename it to the specified ref
+	main, err := git.Reference(DefaultMainReferenceName, false)
+	switch err {
+	case nil: // found it; create the target branch
```

Comment:
Supernit: I feel like go style encourages handling the errors and then not indenting the normal case, but not a blocker

Reply:
oh, ok. easy fix.